litejpeg/core/quantization.py

```diff
@@ -1,42 +1,182 @@
+
+        for i in range(64):
+            self.comb += inverse[i].eq(int((2**16)/quant_values[i]))
+
+
+
+        #Collecting the input values from the previous module and store them in a memory.
+        data_mem = Memory(12, 64*2)
+        data_write_port = data_mem.get_port(write_capable=True)
+        data_read_port = data_mem.get_port(async_read=True)
+        self.specials += data_mem, data_write_port, data_read_port
+
+        write_sel = Signal()
+        write_swap = Signal()
+        read_sel = Signal(reset=1)
+        read_swap = Signal()
+        self.sync += [
+            If(write_swap,
+                write_sel.eq(~write_sel)
+            ),
+            If(read_swap,
+                read_sel.eq(~read_sel)
+            )
         ]
 
+        # write path
+        write_clr = Signal()
+        write_inc = Signal()
+        write_count = Signal(6)
+        self.sync += \
+            If(write_clr,
+                write_count.eq(0)
+            ).Elif(write_inc,
+                write_count.eq(write_count + 1)
+            )
+
         self.comb += [
-            read_port.adr.eq(sink.data[:dw_from//2]),
-            read_port.re.eq(self.pipe_ce),
+            data_write_port.adr.eq(write_count),
+            data_write_port.adr[-1].eq(write_sel),
+            data_write_port.dat_w.eq(sink.data),
+            data_write_port.we.eq(sink.valid & sink.ready)
         ]
-        data = Signal(24)
+
+        self.submodules.write_fsm = write_fsm = FSM(reset_state="IDLE")
+        write_fsm.act("IDLE",
+            write_clr.eq(1),
+            If(write_sel != read_sel,
+                NextState("WRITE")
+            )
+        )
+        write_fsm.act("WRITE",
+            sink.ready.eq(1),
+            If(sink.valid,
+                If(write_count == 63,
+                    write_swap.eq(1),
+                    NextState("IDLE")
+                ).Else(
+                    write_inc.eq(1)
+                )
+            )
+        )
+
+        # read path
+        read_clr = Signal()
+        read_inc = Signal()
+        read_count = Signal(6)
         self.sync += \
-            If(self.pipe_ce,
-                value.eq(sink.data * read_port.dat_r)
-                source.data.eq(data[12:])
+            If(read_clr,
+                read_count.eq(0)
+            ).Elif(read_inc,
+                read_count.eq(read_count + 1)
             )
+
+        
+        # Divider
+        # Here is the code for the dividing the input values with the quantization values.
+
+        # Intialising variables.
+        data_temp_signed = Signal(12)
+        data_temp_unsigned = Signal(12)
+        mult_temp_unsigned = Signal(29)
+        mult_temp_unsigned_original = Signal(12)
+        mult_temp_signed_original = Signal(12)
+        mult_temp_unsigned_round = Signal(12)
+        sign = Signal(1)
+
+
+        self.comb += [
+            # Take an input and store it in data_temp_signed along with the sign.
```

Comment:
The question was "Why are you taking the input and storing it in data_temp_signed along with the sign?" -- IE You are just describing what the code is doing, not *why* it is doing that.